### ISSUE TITLE
CRI: Clearly define the expectation of Summary API and CRI Stats

### ISSUE THREAD

**Issue by Random-Liu** [kubernetes/kubernetes#53514]
In Kubernetes 1.8, we added the support to let kubelet get container stats from CRI.

However, it's not clearly defined whether stats should be returned for non-running container.

Today's situation:
1) Cadvisor integration: it removes stats for terminated containers (containers without cpu/memory usage) https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/stats/cadvisor_stats_provider.go#L93;
2) CRI stats integration: it shows all container stats got from container runtime. It does include stats for terminated container if container runtime returns terminated container stats.

We should:
1) Clearly define the behavior of Summary API in this case.
2) If stats for terminated containers are not required, we should either update kubelet to filter them out, or define in CRI that CRI stats should not include stats with zero memory/cpu usage;
3) If stats for terminated containers are required, we should update Kubelet to properly consume it.

For now, I'll just skip stats with zero memory or cpu usage in cri-containerd.

@timstclair @yujuhong @yguo0905 @dashpole @feiskyer 
/cc @kubernetes/sig-node-api-reviews

**Comment by yguo0905** [comment]
To be precise, we only exclude the **duplicated** terminated container in the cadvisor case.

https://github.com/kubernetes/kubernetes/blob/2eadc737aab12921e6e3a602f2e354862991738a/pkg/kubelet/stats/cadvisor_stats_provider.go#L198-L199

**Comment by Random-Liu** [comment]
@yguo0905 If it only excludes the duplicated terminated container, I feel like this logic should be in kubelet instead of container runtime.

Container runtime should just return stats for all containers.

**Comment by yguo0905** [comment]
Yeah, this logic was added only for fixing the issue https://github.com/kubernetes/kubernetes/issues/47853, where cadvisor reported metrics of uncleaned container cgroups.

**Comment by yujuhong** [comment]
I think providing stats for *all* containers are good. kubelet can filter them out if necessary. cadvisor simply doesn't support that since it only tracks cgroups (for running containers). 


> @yguo0905 If it only excludes the duplicated terminated container, I feel like this logic should be in kubelet instead of container runtime.
> 
> Container runtime should just return stats for all containers.

Excluding duplicated terminated containers is just a workaround for bugs in the existing integration. It was hard to fix for cadvisor+docker because the concept of container in the two components are different.
I don't expect we see duplicated from runtimes integrated via CRI. If we did, it'd be a bug that we need to fix in the specific CRI shim.

**Comment by Random-Liu** [comment]
@yujuhong Thanks. Then based on the above discussion I'll close https://github.com/kubernetes-incubator/cri-containerd/pull/330 and update kubelet instead. :)

Thanks!

**Comment by Random-Liu** [comment]
> I don't expect we see duplicated from runtimes integrated via CRI. If we did, it'd be a bug that we need to fix in the specific CRI shim.

What do you mean "duplicated"? Say we have a container which has been restarted once.
Currently, CRI stats will return stats for both - the newly created one and the terminated one.
Is this duplication in our opinion? /cc @yguo0905
